### PR TITLE
Updating to tensorflow 2.13 and Macos arm issues

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address?
#2158 

##### What are the significant changes in functionality due to this change request?
Updates to tensorflow 2.13, netcdf4 1.6, and makes smt optional.  Also loosens several error checking boundaries because arm floating point is a bit different than x86_64 floating point. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by joshua-cogliati-inl** [comment]
Hm, It looks like decay and lr need to be replaced with LearningRateSchedule https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/LearningRateSchedule
so the lines in ravenframework/SupervisedLearning/KerasBase.py  need to be updated:
```
    if optNode is None:
      optimizerSetting = {'optimizer':'adam'}
    else:
      for sub in optNode.subparts:
        optimizerSetting[sub.getName()] = sub.value
    optimizerName = optimizerSetting.pop('optimizer').lower()
    # set up optimizer
    self.optimizer = self.__class__.availOptimizer[optimizerName](**optimizerSetting)
```

File: dependencies.xml [hunk 1]
```diff
@@ -64,13 +64,11 @@ Note all install methods after "main" take
...
     <dask source="pip" pip_extra="[complete]"/>
-    <ray source="pip" pip_extra="[default]" os='mac,linux'>2.2</ray>
-    <ray source="pip" pip_extra="[default]" os='windows'>1.13</ray>
-    <pydantic source="pip" os='mac,linux'>1</pydantic> <!-- ray 2.2 can't use pydantic 2 -->
+    <ray source="pip" pip_extra="[default]">2.6</ray>
     <!-- redis is needed by ray, but on windows, this seems to need to be explicitly stated -->
     <redis source="pip" os='windows'/>
     <imageio source="pip">2.22</imageio>
-    <smt/>
+    <smt optional='True'/> <!-- not available on macos arm64 -->
```
Comment (wangcj05) [inline dependencies.xml:71]: The code changes are good to me. However I do have a concern about BayesianOptimization tests. I guess we have only one machine that is forced to install all optional libraries. In this case, we will not test the Bayesian Optimization in other machines. I would suggest to at least enable optional libraries installations for MAC, Windows and Sawtooth. @joshua-cogliati-inl Please let me know what do you think.

Comment (joshua-cogliati-inl) [reply dependencies.xml:71]: Hm, installing optional libraries will slow down things by a bit.

Comment (joshua-cogliati-inl) [reply dependencies.xml:71]: Maybe we should add an `machine='x86_64'` or similar that only installs based on what `platform.machine()` returns.

Comment (wangcj05) [reply dependencies.xml:71]: I like the idea, either add another attribute for dependencies or extend optional to accept values that indicate machine info.

Comment (joshua-cogliati-inl) [reply dependencies.xml]: I added an attribute for the dependencies.

File: tests/framework/Optimizers/GeneticAlgorithms/tests [hunk 2]
```diff
@@ -310,7 +310,7 @@
     [./csv]
      type = OrderedCSV
      output = 'continuous/constrained/simionescuConstrainedInvLin/opt_export_0.csv'
-     rel_err = 0.001
+     rel_err = 10.0
```
Comment (wangcj05) [inline tests/framework/Optimizers/GeneticAlgorithms/tests:313]: The relative error here is real big, do you know the reason for this?

File: dependencies.xml [hunk 3]
```diff
@@ -64,13 +65,11 @@ Note all install methods after "main" take
...
     <dask source="pip" pip_extra="[complete]"/>
-    <ray source="pip" pip_extra="[default]" os='mac,linux'>2.2</ray>
-    <ray source="pip" pip_extra="[default]" os='windows'>1.13</ray>
-    <pydantic source="pip" os='mac,linux'>1</pydantic> <!-- ray 2.2 can't use pydantic 2 -->
+    <ray source="pip" pip_extra="[default]">2.6</ray>
     <!-- redis is needed by ray, but on windows, this seems to need to be explicitly stated -->
     <redis source="pip" os='windows'/>
     <imageio source="pip">2.22</imageio>
-    <smt/>
+    <smt machine='x86_64'/> <!-- not available on macos arm64 -->
```
Comment (wangcj05) [inline dependencies.xml:72]: Does it cover all our regression machines?

**Review by wangcj05 — COMMENTED** [review]
I have some questions, could you try to address it?

File: tests/framework/Optimizers/GeneticAlgorithms/tests (hunk 2 above)
Comment (joshua-cogliati-inl) [reply tests/framework/Optimizers/GeneticAlgorithms/tests]: I am not sure why it got so big, so I reverted this change.

File: dependencies.xml (hunk 3 above)
Comment (joshua-cogliati-inl) [reply dependencies.xml:72]: Windows uses AMD64, so I added a bit more to the checking code.

**Review by wangcj05 — APPROVED** [review]
changes are good. Please draft an email for this PR, since we have updated several libraries, and the support for M1/M2 machine. @joshua-cogliati-inl

**Comment by wangcj05** [comment]
The issue will be kept open for future  boost update. The checklist for this PR is satisfied.

**Pull request merged**